src/main/java/org/springframework/integration/kafka/dsl/KafkaProducerMessageHandlerSpec.java

```diff
@@ -258,6 +260,24 @@ public S sendTimeout(long sendTimeout) {
 	}
 
 	/**
+	 * Add a default header mapper to map spring messaging headers to Kafka headers.
+	 * @return the spec.
+	 */
+	public S headerMapper() {
+		return headerMapper(new DefaultKafkaHeaderMapper());
```

Comment:
We don't need this method - see `KafkaProducerMessageHandler` ctor: https://github.com/spring-projects/spring-integration-kafka/blob/master/src/main/java/org/springframework/integration/kafka/outbound/KafkaProducerMessageHandler.java#L95

Reply:
Doh - yes; in my test app the handler was null, so I added one with

```java
		KafkaProducerMessageHandler<Object, Object> messageHandler = Kafka.outboundChannelAdapter(template).topic("rjugIntOut")
				.get();
		messageHandler.setHeaderMapper(new DefaultKafkaHeaderMapper());
		return IntegrationFlows.from(Kafka.messageDrivenChannelAdapter(consumerFactory, containerProperties))
```

and got the no jackson on CP error 😄